great_expectations/rule_based_profiler/types/data_assistant_result/volume_data_assistant_result.py

```diff
@@ -193,59 +191,76 @@ def _create_chart_for_column_domain_expectations(
             for domain in list(attributed_metrics.keys())
         }
 
-        column_dfs: List[Tuple[str, pd.DataFrame]] = []
-        for expectation_configuration in expectation_configurations:
-            metric_configuration: dict = expectation_configuration.meta[
-                "profiler_details"
-            ]["metric_configuration"]
-            domain_kwargs: dict = metric_configuration["domain_kwargs"]
-
-            domain = domains_by_column_name[domain_kwargs["column"]]
+        metric_configuration: dict = expectation_configuration.meta["profiler_details"][
+            "metric_configuration"
+        ]
+        domain_kwargs: dict = metric_configuration["domain_kwargs"]
 
-            attributed_values_by_metric_name: Dict[
-                str, ParameterNode
-            ] = attributed_metrics[domain]
+        domain = domains_by_column_name[domain_kwargs["column"]]
 
-            # Altair does not accept periods.
-            metric_name = list(attributed_values_by_metric_name.keys())[0].replace(
-                ".", "_"
-            )
+        attributed_values_by_metric_name: Dict[str, ParameterNode] = attributed_metrics[
+            domain
+        ]
 
-            df: pd.DataFrame = VolumeDataAssistantResult._create_df_for_charting(
-                metric_name, domain_name, attributed_values_by_metric_name
-            )
+        # Altair does not accept periods.
+        metric_name: str = list(attributed_values_by_metric_name.keys())[0].replace(
+            ".", "_"
+        )
 
-            if prescriptive:
-                for kwarg_name in expectation_configuration.kwargs:
-                    df[kwarg_name] = expectation_configuration.kwargs[kwarg_name]
+        df: pd.DataFrame = VolumeDataAssistantResult._create_df_for_charting(
+            metric_name=metric_name,
+            domain_name=domain_name,
+            attributed_values_by_metric_name=attributed_values_by_metric_name,
+            expectation_configuration=expectation_configuration,
+            prescriptive=prescriptive,
+        )
 
-            column_name: str = expectation_configuration.kwargs["column"]
-            column_dfs.append((column_name, df))
+        column_name: str = expectation_configuration.kwargs["column"]
+        subtitle = f"Column: {column_name}"
 
-        assert metric_name is not None
+        return self._chart_values(
+            df=df,
+            metric_name=metric_name,
+            metric_type=metric_type,
+            domain_name=domain_name,
+            domain_type=domain_type,
+            prescriptive=prescriptive,
+            subtitle=subtitle,
+        )
 
+    def _chart_values(
+        self,
+        df: pd.DataFrame,
+        metric_name: str,
+        metric_type: alt.StandardType,
+        domain_name: str,
+        domain_type: alt.StandardType,
```

Comment:
Thought for a later PR - could we make these `namedtuples`? I think the pairing makes sense since they're always supplied together and it would cut down on the signature size.

Reply:
Sounds reasonable to me!

Reply:
Sounds reasonable to me!

Reply:
I think we could take it a step further and make these objects, encapsulating behavior to produce tool tips and titles for metrics and domains